src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Migration.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\Doctrine\Transport;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+
+final class Migration
+{
+    public static function up(Schema $schema, string $tableName = 'messenger_messages'): void
+    {
+        $useDeprecatedConstants = !class_exists(Types::class);
+
+        $table = $schema->createTable($tableName);
+        $table->addColumn('id', $useDeprecatedConstants ? Type::BIGINT : Types::BIGINT)
+            ->setAutoincrement(true)
+            ->setNotnull(true);
+        $table->addColumn('body', $useDeprecatedConstants ? Type::TEXT : Types::TEXT)
+            ->setNotnull(true);
+        $table->addColumn('headers', $useDeprecatedConstants ? Type::TEXT : Types::TEXT)
+            ->setNotnull(true);
+        $table->addColumn('queue_name', $useDeprecatedConstants ? Type::STRING : Types::STRING)
+            ->setNotnull(true);
+        $table->addColumn('created_at', $useDeprecatedConstants ? Type::DATETIME : Types::DATETIME_MUTABLE)
+            ->setNotnull(true);
+        $table->addColumn('available_at', $useDeprecatedConstants ? Type::DATETIME : Types::DATETIME_MUTABLE)
+            ->setNotnull(true);
+        $table->addColumn('delivered_at', $useDeprecatedConstants ? Type::DATETIME : Types::DATETIME_MUTABLE)
+            ->setNotnull(false);
+        $table->setPrimaryKey(['id']);
+        $table->addIndex(['queue_name']);
+        $table->addIndex(['available_at']);
+        $table->addIndex(['delivered_at']);
```

Comment:
We should also move this code block here: https://github.com/symfony/symfony/blob/0b8adc99a7094bb1b604644ab20c6b1af40abb71/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/PostgreSqlConnection.php#L86-L105

Reply:
Done